hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/handshakeserver/HandshakeServer.java

```diff
@@ -17,51 +27,107 @@
+		}
+	}
 
 	/**
-	 * This function accepts connection from client , calls handshake method
-	 * and if handshake is successful, performs HL7 transaction
-	 * if Handshake fails then it sends a error response to client
+	 * This function accepts connection from client , calls handshake method and if
+	 * handshake is successful, performs HL7 transaction if Handshake fails then it
+	 * sends a error response to client
 	 */
 	private void initConnectionHandler() {
 
-		// Run the ServerSocket in it's own Thread so that it's not blocking the rest of Camel
+		// Run the ServerSocket in it's own Thread so that it's not blocking the rest of
+		// Camel
 		Runnable serverTask = new Runnable() {
 
 			@Override
 			public void run() {
-				ThreadPoolExecutor executor = new ThreadPoolExecutor(properties.getThreadPoolSize(), properties.getThreadPoolSize(),
-                        0L, TimeUnit.MILLISECONDS,
-                        new LinkedBlockingQueue<Runnable>());
-				
+				String methodName = LoggingUtil.getMethodName();
+				ThreadPoolExecutor executor = new ThreadPoolExecutor(properties.getThreadPoolSize(), properties.getThreadPoolSize(), 0L,
+						TimeUnit.MILLISECONDS, new LinkedBlockingQueue<Runnable>());
+
 				ServerSocket mysocket = null;
-				try {					
+				try {
 					mysocket = new ServerSocket(properties.getServerSocket());
 					while (true) {
 						Socket connectionSocket = mysocket.accept();
-						logger.info("{} - Accepting connection attempt from IP Address: {}", LoggingUtil.getMethodName(),
-								connectionSocket.getInetAddress().getCanonicalHostName());
-						
-						ConnectionHandler handler = new ConnectionHandler(producer, connectionSocket, properties.getSocketReadSleepTime(), properties.getMaxSocketReadTries());
-						
+
+						String hostAddress = connectionSocket.getInetAddress().getHostAddress();
+
+						logger.info("{} - Accepting connection attempt from IP Address: {}", methodName,
+								hostAddress);
+
+						// Local connections are always allowed so only validate remote connections
+						if (!isLocalConnection(hostAddress)) {
+							if (!properties.getAcceptRemoteConnections()) {
+								logger.warn("{} - Remote connection rejected: Remote connections not allowed", methodName);
+								continue;
+							} else {
+								if (!isValidIp(hostAddress)) {
+									logger.warn("{} - Remote connection rejected. IP address {} not in valid list", methodName, hostAddress);
+									continue;	
+								}
+							}
+						}
+
+						ConnectionHandler handler = new ConnectionHandler(producer, connectionSocket, properties.getSocketReadSleepTime(),
+								properties.getMaxSocketReadTries());
+
 						Integer activeCount = executor.getActiveCount();
-						
+
 						logger.debug("{} threads are active", activeCount);
 						if (activeCount == properties.getThreadPoolSize()) {
```

Comment:
Not a change in this PR but I just noticed it, should this be .equals() for two Integer objects

Reply:
Switched to use Object.equals() for additional null safety.